Backpack/src/main/java/net/skyscanner/backpack/button/BpkButton.kt

```diff
@@ -25,8 +26,26 @@ open class BpkButton @JvmOverloads constructor(
   defStyleAttr: Int = 0
 ) : AppCompatButton(context, attrs, defStyleAttr) {
 
+  @IntDef(START, END, ICON_ONLY)
+
+  annotation class IconPosition
+
+  companion object {
+    const val START = 0
+    const val END = 1
+    const val ICON_ONLY = 2
+  }
+
+  @BpkButton.IconPosition
+  var iconPosition: Int = BpkButton.END
+    set(value) {
+      field = value
+      this.setup()
+    }
+
+
   private val roundedButtonCorner by lazy(LazyThreadSafetyMode.NONE) { dpToPx(24, context).toFloat() }
-  private val strokeWidth by lazy(LazyThreadSafetyMode.NONE) { 2 }
+  private val strokeWidth by lazy(LazyThreadSafetyMode.NONE) { 4 }
```

Comment:
Are we 100% sure this will never be shared by more than one thread? I'm not sure about how android works internally but if there's any chance this will be shared by more than one thread we shouldn't do this. Also, if we do I think we should add a decision about this, so every lazy field inside a component uses no synchroniser.

Reply:
The default lazy initialization is synchronized which may cause accidental deadlocks. None uses no locks, and therefore it is faster. As we are simply returning primitive values and no complex object it is completely safe and faster to use this than the synchronized one.

Reply:
I understand what you mean but the whole point of lazy is to delay heavy operations and make sure you do them once (because they are heavy). To be honest looking at it again there's no point in having 4 as a lazy value... Maybe we should just remove from roundedButtonCorner as well, as it's not an expensive operation. 

I'm still not sure about the deadlock part as it is supposed to be safe even with the lock, the only diference is that without the lock it _may_  be initialised twice in case this instance is shared by two (or more) threads. 

I think we should just remove the lazy initialisation in this case.

Reply:
I can partially agree with what you say, but if we use tokens here (or at least on roundedButtonCorner we should keep them - as you don't want to call and load values from resources until they are not used/called). On the ones where we have simple values we can remove lazy.